### PR TITLE
Closes #2455: added examples to derive_var_trtemfl (but no diagrams)

### LINKED ISSUE #2455: Documentation:  diagrams for `derive_var_trtemfl`
### Please select a category the issue is focused on?

Function Documentation

### Let us know where something needs a refresh or put your idea here!

Clean up the following diagram and place it in the [`derive_var_trtemfl`](https://pharmaverse.github.io/admiral/reference/derive_var_trtemfl.html) documentation
```mermaid 
flowchart TB

AEITOXGR["AEITOXGR"] -->|"missing AEITOXGR: \n AE not exist before TRTSDT"| arise1["New AE records \n within TEAE window"] --> |New|TEAE[Step 3: Flag records identified above as TEAE]

AEITOXGR -->|"non-missing AEITOXGR:  \n AE exist before TRTSDT"|Worsen1["Step 1: Identify worsened records \n (compare with the previous record) \nafter TRTSDT"]--> Worsen2["Step 2: Get the first worsen date \n after TRTSDT"]-->worsen3["All records after \n the 1st worsen date \n and within TEAE window"]--> |Worsen|TEAE

style TEAE fill:#b3ffb3,stroke:#4CAF50,stroke-width:2px
style Worsen1 fill:#b3ffb3,stroke:#4CAF50,stroke-width:2px
style Worsen2 fill:#b3ffb3,stroke:#4CAF50,stroke-width:2px
```

### PR DESCRIPTION
**Pull request by manciniedoardo**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
I had a look at creating diagrams/flowcarts for `derive_var_trtemfl()` (see for example this [partial flow chart](https://lucid.app/lucidchart/28e9cf4e-e0a6-4eb7-9ca5-0f1902ddf031/edit?page=0_0#). A while into the endeavour i figured that actually it wouldn't be more readable than the bullet-pointed list we have in the reference page. So, instead I just added another example and linked the phuse white paper. The examples section now is a little bit long and has some overlap, but it could be tidied up with our new way of formatting examples. Thoughts @bundfussr @bms63?

**Review by bms63 — COMMENTED** [review]
If diagram doesn't add anything then lets drop it like you suggested!!

Should we make a unit test for these new scenarios just to back up our programming for these scenarios?  maybe the example can just be ported to the unit tests?

**Comment by manciniedoardo** [comment]
> If diagram doesn't add anything then lets drop it like you suggested!!
> 
> Should we make a unit test for these new scenarios just to back up our programming for these scenarios? maybe the example can just be ported to the unit tests?

That's a good idea - added

**Comment by manciniedoardo** [comment]
Not sure of the best way to fix the linting as it's in the roxygen and not the body of the function so I don't think i can `#nolint` it...

**Comment by bundfussr** [comment]
> Not sure of the best way to fix the linting as it's in the roxygen and not the body of the function so I don't think i can `#nolint` it...

@manciniedoardo , at the moment the example is hard to read. For the input dataset not all columns are visible without using the scroll bar (which is unfortunately at the end of the code block):
image
And for the output dataset the new column is not displayed:
image
Removing the times could help. It could also solve the lintr issue.

**Comment by manciniedoardo** [comment]
Thanks @bundfussr I've done that and it looks a lot better in the rendered page - just a tiny bit of scrolling required.

image

**Review by bundfussr — APPROVED** [review]

**Comment by manciniedoardo** [comment]
Only linting problem is now the link. Maybe we should use something like tinyurl

File: R/derive_var_trtemfl.R [hunk 1]
```diff
@@ -117,6 +117,11 @@
 #'
 #'  - Otherwise it is set to `NA_character_`.
 #'
+#'  The behavior of `derive_var_trtemfl` is aligned with the proposed
+#'  treatment-emergent AE assignment in the following
+#'  [PHUSE White Paper](https://phuse.s3.eu-central-1.amazonaws.com/Deliverables/Safety+Analytics/WP-087+Recommended+Definition+of++Treatment-Emergent+Adverse+Events+in+Clinical+Trials+.pdf).
```
Comment (bundfussr) [inline R/derive_var_trtemfl.R:122]:
```suggestion
# nolint start
#'  [PHUSE White Paper](https://phuse.s3.eu-central-1.amazonaws.com/Deliverables/Safety+Analytics/WP-087+Recommended+Definition+of++Treatment-Emergent+Adverse+Events+in+Clinical+Trials+.pdf).
# nolint end
```
This should solve the lintr issue.

Comment (bundfussr) [inline R/derive_var_trtemfl.R:122]:
```suggestion
# nolint start
#'  [PHUSE White Paper](https://phuse.s3.eu-central-1.amazonaws.com/Deliverables/Safety+Analytics/WP-087+Recommended+Definition+of++Treatment-Emergent+Adverse+Events+in+Clinical+Trials+.pdf).
# nolint end
```
This should solve the lintr issue.

Comment (bms63) [reply R/derive_var_trtemfl.R:123]: will this appear in the documentation/website.   The nolint stuff?

Comment (manciniedoardo) [reply R/derive_var_trtemfl.R:123]: I think the intention is no since we are using `#` and not `#'`

Comment (bms63) [reply R/derive_var_trtemfl.R:123]: I feel like this has happened before...let me render the site.

I wish we had a deploy dev site feature - maybe somethign we can add??

Comment (bundfussr) [reply R/derive_var_trtemfl.R:123]: @bms63 , you could just look at the .Rd file in `man`.
